### PR TITLE
modified EKI/EKS runtest

### PR DESCRIPTION
**Pull request by odunbar**
## Purpose

To get the EKI runtest properly working with a linear map from R^{2} -> R^{10}. (i) some reason a couple of the `@test` statements were commented out (ii) the tests comparing EKS and EKI are very dependent on N_iter

## In the PR
Changes to the EKI test in `test/EnsembleKalmanInversion/runtests.jl` So that it now converges reasonably and satisfies all the test conditions. 

NB some of the comparisons of eks/eki depended heavily on `N_iter`. To compare the two algorithms we should really look at the timesteps in absolute time. So now
1. I have increased the prior strength and bias
1. I run EKS first with it's adaptive timestep, which I store in `eksobj`
2.  then use these values in the EKI updates. 

The OLS mean is nearer the EKI at the end time, and the posterior mean is nearer the EKS at the end time, far more robustly than previously. 

## Test output plot
For EKI
EKI_test

For EKS
EKS_test

### CODE REVIEW

**Comment by odunbar** [comment]
@agarbuno  @jinlong83 could I get some reviews on this please

File: test/EnsembleKalmanProcessModule/runtests.jl [hunk 1]
```diff
@@ -8,65 +8,111 @@ using EnsembleKalmanProcesses.ParameterDistributionStorage
...
+    end
+    # Plot evolution of the EKS particles
+    eks_final_result = vec(mean(get_u_final(eksobj), dims = 2))
+
+    if TEST_PLOT_OUTPUT
+        gr()
+        p = plot(get_u_prior(eksobj)[1, :], get_u_prior(eksobj)[2, :], seriestype = :scatter)
+        plot!(get_u_final(eksobj)[1, :], get_u_final(eksobj)[2, :], seriestype = :scatter)
+        plot!([u_star[1]], xaxis = "u1", yaxis = "u2", seriestype = "vline", linestyle = :dash, linecolor = :red)
+        plot!([u_star[2]], seriestype = "hline", linestyle = :dash, linecolor = :red)
```
Comment (navidcy) [inline test/EnsembleKalmanProcessModule/runtests.jl:90]:
```suggestion
        plot!([u_star[2]], seriestype = "hline", linestyle = :dash, linecolor = :red, label= :none)
```

File: test/EnsembleKalmanProcessModule/runtests.jl [hunk 2]
```diff
@@ -8,65 +8,111 @@ using EnsembleKalmanProcesses.ParameterDistributionStorage
...
+        EnsembleKalmanProcessModule.update_ensemble!(eksobj, g_ens)
+    end
+    # Plot evolution of the EKS particles
+    eks_final_result = vec(mean(get_u_final(eksobj), dims = 2))
+
+    if TEST_PLOT_OUTPUT
+        gr()
+        p = plot(get_u_prior(eksobj)[1, :], get_u_prior(eksobj)[2, :], seriestype = :scatter)
+        plot!(get_u_final(eksobj)[1, :], get_u_final(eksobj)[2, :], seriestype = :scatter)
+        plot!([u_star[1]], xaxis = "u1", yaxis = "u2", seriestype = "vline", linestyle = :dash, linecolor = :red)
```
Comment (navidcy) [inline test/EnsembleKalmanProcessModule/runtests.jl:89]:
```suggestion
        plot!([u_star[1]], xaxis = "u1", yaxis = "u2", seriestype = "vline", linestyle = :dash, linecolor = :red, label = :none)
```

File: test/EnsembleKalmanProcessModule/runtests.jl [hunk 3]
```diff
@@ -8,65 +8,111 @@ using EnsembleKalmanProcesses.ParameterDistributionStorage
...
+
+        EnsembleKalmanProcessModule.update_ensemble!(eksobj, g_ens)
+    end
+    # Plot evolution of the EKS particles
+    eks_final_result = vec(mean(get_u_final(eksobj), dims = 2))
+
+    if TEST_PLOT_OUTPUT
+        gr()
+        p = plot(get_u_prior(eksobj)[1, :], get_u_prior(eksobj)[2, :], seriestype = :scatter)
+        plot!(get_u_final(eksobj)[1, :], get_u_final(eksobj)[2, :], seriestype = :scatter)
```
Comment (navidcy) [inline test/EnsembleKalmanProcessModule/runtests.jl:88]:
```suggestion
        plot!(get_u_final(eksobj)[1, :], get_u_final(eksobj)[2, :], seriestype = :scatter, label = "final distribution")
```

File: test/EnsembleKalmanProcessModule/runtests.jl [hunk 4]
```diff
@@ -8,65 +8,111 @@ using EnsembleKalmanProcesses.ParameterDistributionStorage
...
+        end
+
+        EnsembleKalmanProcessModule.update_ensemble!(eksobj, g_ens)
+    end
+    # Plot evolution of the EKS particles
+    eks_final_result = vec(mean(get_u_final(eksobj), dims = 2))
+
+    if TEST_PLOT_OUTPUT
+        gr()
+        p = plot(get_u_prior(eksobj)[1, :], get_u_prior(eksobj)[2, :], seriestype = :scatter)
```
Comment (navidcy) [inline test/EnsembleKalmanProcessModule/runtests.jl:87]:
```suggestion
        p = plot(get_u_prior(eksobj)[1, :], get_u_prior(eksobj)[2, :], seriestype = :scatter, label = "initial distribution")
```

**Review by navidcy — COMMENTED** [review]
I made few suggestions for plotting style

**Comment by odunbar** [comment]
Thanks Navid!

**Comment by agarbuno** [comment]
I guess we can't really compare the two algorithms at a specific simulated time `T`. The thing is both EKI and EKS have very different long time dynamics. 

1.  We know EKS stabilizes at some time (there are some theoretical results on it stabilizing in an attractor set of Gaussians) and it stays there which in the long run is the approximate posterior. 
2. We know EKI in the long run collapses to the solution of the Least Square problem. 

So if `N_iter` is sufficiently large then:

1. EKI should collapse to the solution of the OLS in this case. 
2. EKS should be on top of the posterior. 

Maybe try with a sufficiently large number of ensemble updates? Since this is a linear and Gaussian setting it shouldn't take so long and from the figures I guess it has been sufficient for both methods to stabilize in their respective long-time behavior.

**Review by agarbuno — APPROVED** [review]

**Comment by odunbar** [comment]
Thanks!

In the previous implementation, different realizations, steps and ensemble sizes caused tests to fail. I think that this test gives more separation between the OLS and posterior mean. and more reliable results.

We can discuss offline about the algorithmic time implication of how comparable the progression of the algorithms are.

**Pull request merged**